### ISSUE TITLE
dangers of magic

### ISSUE THREAD

**Issue by rand99** [jinzhu/gorm#792]
I just saw something interesting while developing.
I wanted to set "payed" = 1 to the user.
According to the docs this should do the trick
```
// Update an attribute if it is changed
db.Model(&user).Update("name", "hello")
//// UPDATE users SET name='hello', updated_at = '2013-11-17 21:34:10' WHERE id=111;
```
but my code somehow resulted in this
```
u := &User{ID:user_id}
res:=DBcon.Model(u).Update("Payed", 1)

[2016-01-09 17:08:25]  [0.69ms]  UPDATE "users" SET 
"ip" = '', 
"created_at" = '0001-01-01T00:00:00Z', 
"updated_at" = '2016-01-09T17:08:25+01:00',
 "pw" = '', 
"payed" = '1',
 "lang" = ''  WHERE ("id" = '67055b54-e51e-4021-b405-02e96bbfa47e')

```

So if a customer pays his password gets zero and everybody is welcome to login.
This might be a bug or something with the docs?

[update]

further test show that the & changes a lot
```
res:=DBcon.Model(&User{ID:user_id}).Update("Payed", 1)
[2016-01-09 17:29:32]  [1.04ms]  UPDATE "users" SET "pw" = '', "payed" = '1', "lang" = '',
 "ip" = '', "created_at" = '0001-01-01T00:00:00Z', "updated_at" = '2016-01-09T17:29:32+01:00'  
WHERE ("id" = '67055b54-e51e-4021-b405-02e96bbfa47e')

res:=DBcon.Model(User{ID:user_id}).Update("Payed", 1)
[2016-01-09 17:31:34]  [0.88ms]  UPDATE "users" SET "payed" = '1'  WHERE ("id" = '67055b54-e51e-4021-b405-02e96bbfa47e')
```

**Comment by Gacnt** [comment]
I think in the documentation, `&user` is supposed to be a result from a database that has been selected, not one that is newly allocated like your `u := &User{ID:user_id} // New type &User` so when you update your field in your element, you're updating it with all the other `zeroed` values of the new `User` type. So to avoid doing this I would recommend updating like:

```
db.Model(model.Characters{}).Where("id = ?", user_id).Updates(map[string]interface{}{"character_main": 1})
```

**Comment by jinzhu** [comment]
Yes, although I am thinking how to improve this, going to make some changes in the v1.0 to fix this issue.

**Comment by galeone** [comment]
@jinzhu my PR fixes this issue.

**Comment by jinzhu** [comment]
Hi @galeone 

I don't think it does, and I think it is not a right behaviour for me.

When updating some values, it should doesn't matter to the default value, I should just update the value to what I have set.

I will try to make this part clear in the v1.0 release.

**Comment by jinzhu** [comment]
This should be fixed in v1.0_dev branch.

And added more documents about this in https://github.com/jinzhu/gorm/tree/v1.0_dev#update

Feedback welcome in #794
